src/daemon/daemon.cpp

```diff
@@ -3081,7 +3081,7 @@ mp::Daemon::async_wait_for_ready_all(grpc::ServerReaderWriterInterface<Reply, Re
                         mpu::run_in_ssh_session(session, command);
                     }
                 }
-                catch (const std::runtime_error&) // In case there is an error executing the command, report it.
+                catch (const mp::SSHExecFailure&) // In case there is an error executing the command, report it.
                 {
```

Comment:
It is already a big improvement. Just one small addition, maybe we can also add another branch below to catch `mp::SSHException` and return an error message because there is still a chance that the `mp::SSHSession` constructor throws.

Reply:
Hey @georgeliao, thanks! That is a good idea but, what to do in the `catch SSHException&` branch? Just message a warning?

Reply:
If that fails, then it can be still a warning, right? We suggest user to do it manually.

Reply:
Ok, but in that case it will be the same as before...

Reply:
Or almost

Reply:
Hmm, indeed, then just catching mp::SSHException alone and append the exception message can be an option as well.

Reply:
Done! Please take a look to the change.

Reply:
Great, it looks way better now. just one small error message wording improvement.